corehq/apps/app_manager/models.py

```diff
@@ -5414,8 +5415,20 @@ def get_case_types(self):
         extra_types = set()
         if is_usercase_in_use(self.domain):
             extra_types.add(USERCASE_TYPE)
+
         return set(chain(*[m.get_case_types() for m in self.get_modules()])) | extra_types
 
+    @memoized
+    def get_shared_case_types(self):
```

Comment:
I was originally going to add this to `get_case_types` but had some issues with tests (I think it was just due to the mocking used in some of the tests though).

Where do you think the best place for this would be?